pkg/ovssubnet/controller/kube/bin/openshift-sdn-kube-subnet-setup.sh

```diff
@@ -25,39 +25,18 @@ function docker_network_config() {
 	DOCKER_NETWORK_OPTIONS="-b=lbr0 --mtu=${mtu}"
     fi
 
+    local conf=/run/openshift-sdn/docker-network
     case "$1" in
 	check)
-	    if [ -f /.dockerinit ]; then
-		# Assume supervisord-managed docker for docker-in-docker deployments
-		conf=/etc/supervisord.conf
-		if ! grep -q -s "DOCKER_DAEMON_ARGS=\"${DOCKER_NETWORK_OPTIONS}\"" $conf; then
-		    return 1
-		fi
-	    else
-		# Otherwise assume systemd-managed docker
-		conf=/run/openshift-sdn/docker-network
-		if ! grep -q -s "DOCKER_NETWORK_OPTIONS='${DOCKER_NETWORK_OPTIONS}'" $conf; then
-		    return 1
-		fi
+	    if ! grep -q -s "DOCKER_NETWORK_OPTIONS='${DOCKER_NETWORK_OPTIONS}'" $conf; then
+		return 1
 	    fi
 	    return 0
 	    ;;
 
 	update)
-	    if [ -f /.dockerinit ]; then
-		conf=/etc/supervisord.conf
-		if [ ! -f $conf ]; then
-		    echo "Running in docker but /etc/supervisord.conf not found." >&2
-		    exit 1
-		fi
-
-		echo "Docker networking options have changed; manual restart required." >&2
-		sed -i.bak -e \
-		    "s+\(DOCKER_DAEMON_ARGS=\)\"\"+\1\"${DOCKER_NETWORK_OPTIONS}\"+" \
-		    $conf
-	    else
 		mkdir -p /run/openshift-sdn
-		cat <<EOF > /run/openshift-sdn/docker-network
+		cat <<EOF > $conf
```

Comment:
not sure we really need $conf still, but if you're keeping it, the mkdir should be `mkdir -p $(dirname $conf)`